spec/unit/filter_spec.rb

```diff
@@ -122,4 +122,57 @@
       expect(filter.matches?("/changed_managed_files/files/name", "/var/cache")).to be(false)
     end
   end
+
+  describe "#apply!" do
+    it "filters simple elements" do
+      description = create_test_description(scopes: ["unmanaged_files"])
+
+      expect(description.unmanaged_files.files.map(&:name)).to match_array([
+        "/etc/unmanaged-file",
+        "/etc/another-unmanaged-file"
+      ])
+
+      filter = Filter.new("/unmanaged_files/files/name=/etc/unmanaged-file")
+      filter.apply!(description)
+
+      expect(description.unmanaged_files.files.map(&:name)).to match_array([
+        "/etc/another-unmanaged-file"
+      ])
+    end
+
+    it "filters by array matches" do
+      description = create_test_description(scopes: ["changed_managed_files"])
+
+      expect(description.changed_managed_files.files.map(&:name)).to match_array([
+        "/etc/deleted changed managed",
```

Comment:
Would it be possible to express that in a way which resembles the system description a bit closer? It's hard to understand the test without context.